File: backend/notifications/src/main/java/com/oxygensend/notifications/domain/PhoneCode.java
package com.oxygensend.notifications.domain;

import java.util.Arrays;
import java.util.Objects;

public enum PhoneCode {

    POLAND("+48", "PL"),
    GERMANY("+49", "DE"),
    FRANCE("+33", "FR"),
    UNITED_KINGDOM("+44", "GB"),
    UNITED_STATES("+1", "US");

    public final String code;
    public final String country;

    PhoneCode(String code, String country) {
        this.code = code;
        this.country = country;
    }

    public static PhoneCode fromCountry(String country) {
        return Arrays.stream(PhoneCode.values())
                     .filter(phoneCode -> Objects.equals(phoneCode.country, country))
                     .findFirst()
                     .orElseThrow();
    }

}

File: src/main/kotlin/com/oxygensend/notifications/domain/recipient/PhoneCode.kt
package com.oxygensend.notifications.domain.recipient

import java.util.*

enum class PhoneCode(val code: String, val country: String) {
    POLAND("+48", "PL"),
    GERMANY("+49", "DE"),
    FRANCE("+33", "FR"),
    UNITED_KINGDOM("+44", "GB"),
    UNITED_STATES("+1", "US");

    companion object {
        fun fromCountry(country: String?): PhoneCode {
            return Arrays.stream(entries.toTypedArray())
                .filter { phoneCode: PhoneCode -> phoneCode.country == country }
                .findFirst()
                .orElseThrow()
        }
    }
}
